### PR TITLE
Fix errors and omissions in the developer tutorial

### PR DESCRIPTION
**Pull request by settermjd**
There were a series of mistakes in the example code, along with some of the text, that could lead developers to make unnecessary and frustrating mistakes.

### CODE REVIEW

**Comment by mmattel** [comment]
@PVince81 mind to review so we can proceed ?

File: modules/developer_manual/pages/app/tutorial/development_environment.adoc [hunk 1]
```diff
@@ -20,10 +20,24 @@ In addition to these, there are several additional, commonly used,
 directories:
 
 * `css/`: Contains the CSS files
+* `img/`: Contains the image files
 * `js/`: Contains the JavaScript files
 * `templates/`: Contains the templates
 * `tests/`: Contains the tests
```
Comment (PVince81) [inline modules/developer_manual/pages/app/tutorial/development_environment.adoc]: should we also add the "vendor" folder ?
usually there's also a "lib" folder with the classes

File: modules/developer_manual/pages/app/tutorial/development_environment.adoc [hunk 2]
```diff
@@ -50,17 +64,18 @@ In `appinfo/info.xml`, add the following XML, changing it as necessary:
     <namespace>OwnNotes</namespace>
     <category>tool</category>
     <dependencies>
-        <owncloud min-version="9" />
-        <owncloud max-version="10" />
+        <owncloud min-version="10.0.0" max-version="11.0.0.0" />
```
Comment (PVince81) [inline modules/developer_manual/pages/app/tutorial/development_environment.adoc]: max-version="10"

File: modules/developer_manual/pages/app/tutorial/development_environment.adoc (hunk 1 above)
Comment (settermjd) [reply modules/developer_manual/pages/app/tutorial/development_environment.adoc]: I've done a scan and not found any vendor directories yet. However, I did find `bin` and `l10n` directories using [a bash one-liner](https://gist.github.com/settermjd/5c15e1d1683d3f75d077349adaee9718), so have included those.

File: modules/developer_manual/pages/app/tutorial/development_environment.adoc (hunk 2 above)
Comment (settermjd) [reply modules/developer_manual/pages/app/tutorial/development_environment.adoc]: Well spotted. Thanks.

Comment (settermjd) [reply modules/developer_manual/pages/app/tutorial/development_environment.adoc]: Is `<owncloud min-version="9.2.0" max-version="10.0.0" />` suitable?

**Comment by settermjd** [comment]
Feedback has been implemented.

**Pull request merged**